Source/WebCore/loader/cache/CachedFont.cpp

```diff
@@ -134,8 +129,16 @@ String CachedFont::calculateItemInCollection() const
 bool CachedFont::ensureCustomFontData(SharedBuffer* data)
 {
     if (!m_fontCustomPlatformData && !errorOccurred() && !isLoading() && data) {
-        bool wrapping;
-        m_fontCustomPlatformData = createCustomFontData(*data, calculateItemInCollection(), wrapping);
+        bool wrapping = false;
+        if (m_mustLoadWithMemorySafeParser) {
+            if (m_loader->frame()->settings().experimentalFontParserInLockdownMode())
+                m_fontCustomPlatformData = createCustomFontDataExperimentalParser(*data, calculateItemInCollection(), wrapping);
+            if (!m_fontCustomPlatformData) {
+                m_didRefuseToParseCustomFont = true;
+                setErrorAndDeleteData();
```

Comment:
It is just that it is a bit weird for me that we are trying to parse at `CachedFont::ensureCustomFontData(SharedBuffer* data)` and an ensure function can potentially still invalidate loading. At that point I’d expect loading to be concluded already, and parsing would be part of loading.

Notice that `ensureCustomFontData` is called from `CSSFontFaceSource::fontLoaded`

Maybe we can come up with a better abstraction for it? Otherwise is not critical, of course.

Reply:
FWIW, parsing was done in `ensureCustomFontData` before my change (see [138-](https://github.com/WebKit/WebKit/pull/23123/files#diff-47a745755cc9d17fbba3686ad6e0eadc2a4bb514158235a72e81389552fcc04eL138)). I agree that `setErrorAndDeleteData` will probably never actually cancel the m_loader.